presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarAggregationToJoin.java

```diff
@@ -86,12 +87,13 @@ public TransformCorrelatedScalarAggregationToJoin(FunctionRegistry functionRegis
     @Override
     public Result apply(LateralJoinNode lateralJoinNode, Captures captures, TraitSet traitSet, Context context)
     {
-        PlanNode subquery = lateralJoinNode.getSubquery();
-
-        if (!isScalar(subquery, context.getLookup())) {
+        Optional<CardinalityTrait> subqueryCardinality = context.getLookup().resolveTrait(lateralJoinNode.getSubquery(), CARDINALITY);
+        if (!subqueryCardinality.isPresent() || !subqueryCardinality.get().isScalar()) {
             return Result.empty();
         }
 
+        PlanNode subquery = context.getLookup().resolve(lateralJoinNode.getSubquery());
```

Comment:
@kokosing `context.getLookup().resolve` wasn't needed before; why is it necessary now?

Reply:
It is a rebase related mistake. Thanks.